### DISCUSSION TITLE
How to compile a single file using g++ from the command line

### DISCUSSION THREAD

**Discussion by Tzlzl12** [alandefreitas/matplotplusplus#416] [Q&A]
I have installed matplot++
![Snipaste_2024-07-27_12-56-11](https://github.com/user-attachments/assets/ad09e0b4-0d70-4a31-a74e-de01a6bdbbfa)
When I pass g++01.cc he shows
![Snipaste_2024-07-27_12-56-41](https://github.com/user-attachments/assets/2db72418-b57f-4db1-8876-68dd829fd6fa)
So I added a g++01.cc -lmatplot but
![Snipaste_2024-07-27_12-57-07](https://github.com/user-attachments/assets/fad26111-06fa-4ce0-ae42-58af9446a0e3)
Looks like it's getting worse
Therefore, I hope someone can give me a correct compilation command, thank you

**Comment by Tzlzl12** [comment]
The length of the second error message
![Snipaste_2024-07-27_13-02-25](https://github.com/user-attachments/assets/4149d270-e787-439b-859b-7aade1e1f41f)

**Comment by duaneg3** [comment]
I think you might have the order of libraries incorrect, and it is seems best to rebuild the library ..

Rebuilding is easy and quick (works on Ubuntu-20.4)

// not really sure if the first one is needed
sudo apt-get install libglfw3-dev
sudo apt-get install libjpeg-dev
sudo apt-get install libpng-dev
sudo apt-get install gnuplot
sudo apt-get install qt5-default

get the lastest copy of the repository
cd matplotplusplus/
mkdir compile-release
cd compile-release 
cmake .. -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release -DMATPLOTPP_BUILD_EXAMPLES=OFF -DMATPLOTPP_BUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX="./release" 
make -j4 
make install
// make -VERBOSE=1 -j4  might be interesting

// the order of the libraries included on the command line matters!
g++ -std=c++17 test_plot.cpp -I$(MATPLOT)/include -L$(MATPLOT_LIB)/lib -lmatplot -lnodesoup -ljpeg -lpng -lpthread -lz -o test_plot	

In the end, I found the pipe interface to gnuplot to be cumbersome, and decided not to use the library.
I wanted to generate a plot in a non graphic terminal window, and could not get the library to work well.
export GNUTERM="png"
helped, but still resulted in garbage to the screen
